lib/teams/aws-batch/aws-batch-on-eks-team.ts

```diff
@@ -0,0 +1,167 @@
+import { assertEC2NodeGroup } from "../../cluster-providers";
+import { ClusterInfo, Values } from "../../spi";
+import { ApplicationTeam, TeamProps } from "../team";
+import { KubectlProvider, ManifestDeployment } from "../../addons/helm-addon/kubectl-provider";
+import { loadYaml, readYamlDocument } from "../../utils/yaml-utils";
+import { Construct } from "constructs";
+import { createNamespace } from "../../utils";
+
+import * as batch from 'aws-cdk-lib/aws-batch';
+import * as eks from 'aws-cdk-lib/aws-eks';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import { AutoScalingGroup } from "aws-cdk-lib/aws-autoscaling";
+import * as iam from "aws-cdk-lib/aws-iam";
+
+const NAMESPACE = 'aws-batch';
+
+/**
+ * Interface to define an AWS Batch on EKS team
+ */
+export interface BatchEksTeamProps extends TeamProps {
+  /**
+   * Name of the team
+   */
+  name: string,
+
+  /**
+   * Namespace name for AWS Batch
+   */
+  namespace: string
+
+  /**
+   * Allocation strategies for EKS Compute environment
+   */
+  allocationStrategy?: 'BEST_FIT_PROGRESSIVE' | 'SPOT_CAPACITY_OPTIMIZED'
+
+  jobQueueName: string
+
+  priority: number,
+}
+
+const defaultProps: BatchEksTeamProps = {
+  name: NAMESPACE,
+  namespace: NAMESPACE,
+  allocationStrategy: 'BEST_FIT_PROGRESSIVE',
+  jobQueueName: 'batch-eks-job',
+  priority: 10,
+}
+
+/*
+ *This class define the Team that can be used with AWS Batch on EKS
+ */
+
+export class BatchEksTeam extends ApplicationTeam {
+
+  private batchTeam: BatchEksTeamProps;
+  /**
+   * @public
+   * @param {BatchEksTeamProps} props the Batch team definition {@link BatchEksTeamProps}
+   */
+  constructor(props: BatchEksTeamProps) {
+    super({...defaultProps, ...props});
+    this.batchTeam = props;
+  }
+
+  setup(clusterInfo: ClusterInfo): void {
+    const cluster = clusterInfo.cluster;
+    const allocStr = this.batchTeam.allocationStrategy || 'BEST_FIT_PROGRESSIVE';
```

Comment:
why this check?  You set default explicitly, can it ever be unset?

Reply:
addressed